app/client/src/components/pages/MonitoringStation.js

```diff
@@ -0,0 +1,2163 @@
+  parentObj[parentId] = {
+    ...parentObj[parentId],
+    selected: parentSelected,
+  };
+}
+
+function updateSelected(charcs, types, groups) {
+  let groupsSelected = 0;
+
+  // loop over again to get checkbox values
+  Object.values(charcs).forEach((charc) => {
+    types[charc.type].selected += charc.selected;
+  });
+  Object.values(types).forEach((type) => {
+    type.selected = getCheckedStatus(type.selected, type.charcs);
+    groups[type.group].selected += type.selected;
+  });
+  Object.values(groups).forEach((group) => {
+    group.selected = getCheckedStatus(group.selected, group.types);
+    groupsSelected += group.selected;
+  });
+
+  return getCheckedStatus(groupsSelected, Object.keys(groups));
+}
+
+function useCharacteristics(provider, orgId, siteId) {
+  const services = useServicesContext();
+
+  // charcs => characteristics
+  const [charcs, setCharcs] = useState({});
+  const [status, setStatus] = useState('idle');
+
+  const structureRecords = useCallback(
+    (records) => {
+      if (!records) {
+        setCharcs({});
+        setStatus('failure');
+        return;
+      } else if (!records.length) {
+        setCharcs({});
+        setStatus('empty');
+        return;
+      }
+      const recordsByCharc = {};
+      records.forEach((record) => {
+        if (!recordsByCharc[record.CharacteristicName]) {
+          const charcType = getCharcType(
+            record.CharacteristicName,
+            characteristicsByType,
+          );
+          recordsByCharc[record.CharacteristicName] = {
+            name: record.CharacteristicName,
+            group: getCharcGroup(charcType, characteristicGroupMappings),
+            records: [],
+            type: charcType,
+            count: 0,
+          };
+        }
+        const curCharc = recordsByCharc[record.CharacteristicName];
+        curCharc.count += 1;
+        const recordDate = record.ActivityStartDate.split('-');
+        curCharc.records.push({
+          year: parseInt(recordDate[0]),
+          month: parseInt(recordDate[1]),
+          day: parseInt(recordDate[2]),
+          measurement: record.ResultMeasureValue ?? null,
+          sampleFraction: record.ResultSampleFractionText || null,
+          speciation: record.MethodSpecificationName || null,
+          unit: record['ResultMeasure/MeasureUnitCode'] || null,
+        });
+      });
+      setCharcs(recordsByCharc);
+      setStatus('success');
+    },
+    [setCharcs, setStatus],
+  );
+
+  useEffect(() => {
+    if (services.status !== 'success') return;
+    setStatus('fetching');
+    const url =
+      `${services.data.waterQualityPortal.resultSearch}` +
+      `&mimeType=csv&zip=no&dataProfile=narrowResult` +
+      `&providers=${provider}&organization=${orgId}&siteid=${siteId}`;
+    try {
+      fetchParseCsv(url).then((results) => structureRecords(results.data));
+    } catch (err) {
+      setStatus('failure');
+      console.error('Papa Parse error');
+    }
```

Comment:
```suggestion
    fetchParseCsv(url)
      .then((results) => structureRecords(results.data))
      .catch((err) => {
        setStatus('failure');
        console.error('Papa Parse error');      
      });
```

This try/catch isn't working, because `fetchParseCsv` just returns a promise and doesn't throw any exceptions. So if you block the `https://www.waterqualitydata.us/data/Result/search?` url in Chrome, then you just get infinite loading spinners. 

To fix this switch to the then/catch style of logic for this.